richeditor-compose/src/commonMain/kotlin/com/mohamedrejeb/richeditor/model/RichSpanStyle.kt

```diff
@@ -0,0 +1,150 @@
+            if (lastChild == null) break
+            textRange = TextRange(
+                start = textRange.min,
+                end = lastChild.textRange.max
+            )
+        }
+        return textRange
+    }
+
+    val fullSpanStyle: SpanStyle
+        get() {
+        var spanStyle = this.spanStyle
+        var parent = this.parent
+
+        while (parent != null) {
+            spanStyle = parent.spanStyle.merge(spanStyle)
+            parent = parent.parent
+        }
+
+        return spanStyle
+    }
+
+    fun getSpanStyleByTextIndex(
+        textIndex: Int,
+        offset: Int = 0,
+    ): Pair<Int, RichSpanStyle?> {
+        var index = offset
+
+        // Set start text range
+        textRange = TextRange(start = index, end = index + text.length)
+        index += this.text.length
+
+        // Check if the text index is in the start text range
+        if (textIndex in textRange)
+            return index to this
+
+        // Check if the text index is in the children
+        children.forEach { richSpanStyle ->
+            val result = richSpanStyle.getSpanStyleByTextIndex(
+                textIndex = textIndex,
+                offset = index,
+            )
+            if (result.second != null)
+                return result
+            else
+                index = result.first
+        }
+
+//        fullTextRange = TextRange(start = textRange.min, end = index)
+        return index to null
+    }
+
+    fun getSpanStyleListByTextRange(
+        searchTextRange: TextRange,
+        offset: Int = 0,
+    ): Pair<Int, List<RichSpanStyle>> {
+        var index = offset
+        val richSpanStyleList = mutableListOf<RichSpanStyle>()
+
+        // Set start text range
+        textRange = TextRange(start = index, end = index + text.length)
+        index += this.text.length
+
+        // Check if the text index is in the start text range
+        if (searchTextRange.min < textRange.max && searchTextRange.max > textRange.min)
+            richSpanStyleList.add(this)
+
+        // Check if the text index is in the children
+        children.forEach { richSpanStyle ->
+            val result = richSpanStyle.getSpanStyleListByTextRange(
+                searchTextRange = searchTextRange,
+                offset = index,
+            )
+            richSpanStyleList.addAll(result.second)
+            index = result.first
+        }
+
+        return index to richSpanStyleList
+    }
+
+    fun removeTextRange(textRange: TextRange): RichSpanStyle? {
+        if (textRange.min <= this.textRange.min && textRange.max >= fullTextRange.max) return null
+
+        // Remove text from start
+        if (textRange.min in this.textRange || (textRange.max - 1) in this.textRange) {
+            println("Remove text from start")
+            println("textRange: $textRange")
+            println("this.textRange: ${this.textRange}")
+            println("text: ${this.text}")
```

Comment:
I guess these will be removed

Reply:
Yes, it's not ready yet for merge. I'm still adding few things here and there.